File: examples/python/code/v1alpha/text_to_image.py
import os
import requests

engine_id = "stable-diffusion-v1-5"
api_host = os.getenv('API_HOST', 'https://api.stability.ai')
url = f"{api_host}/v1alpha/generation/{engine_id}/text-to-image"
output_file = "./out/v1alpha_txt2img.png"

apiKey = os.getenv("STABILITY_API_KEY")
if apiKey is None:
    raise Exception("Missing Stability API key.")

payload = {
    "cfg_scale": 7,
    "clip_guidance_preset": "FAST_BLUE",
    "height": 512,
    "width": 512,
    "samples": 1,
    "steps": 50,
    "text_prompts": [
        {
            "text": "A lighthouse on a cliff",
            "weight": 1
        }
    ],
}

headers = {
    "Content-Type": "application/json",
    "Accept": "image/png",
    "Authorization": apiKey
}

response = requests.post(url, json=payload, headers=headers)

if response.status_code != 200:
    raise Exception("Non-200 response: " + str(response.text))

# Write the bytes from response.content to a file
with open(output_file, "wb") as f:
    f.write(response.content)

File: examples/go/code/v1alpha/text_to_image.go
package main

import (
	"bytes"
	"io"
	"net/http"
	"os"
)

func main() {
	// Build REST endpoint URL w/ specified engine
	engineId := "stable-diffusion-v1-5"
	apiHost, hasApiHost := os.LookupEnv("API_HOST")
	if !hasApiHost {
		apiHost = "https://api.stability.ai"
	}
	reqUrl := apiHost + "/v1alpha/generation/" + engineId + "/text-to-image"

	// Acquire an API key from the environment
	apiKey, hasAPIKey := os.LookupEnv("STABILITY_API_KEY")
	if !hasAPIKey {
		panic("Missing STABILITY_API_KEY environment variable")
	}

	var data = []byte(`{
		"cfg_scale": 7,
		"clip_guidance_preset": "FAST_BLUE",
		"height": 512,
		"width": 512,
		"samples": 1,
		"steps": 50,
		"text_prompts": [
		  {
			"text": "A lighthouse on a cliff",
			"weight": 1
		  }
		]
  	}`)

	req, _ := http.NewRequest("POST", reqUrl, bytes.NewBuffer(data))
	req.Header.Add("Content-Type", "application/json")
	req.Header.Add("Accept", "image/png")
	req.Header.Add("Authorization", apiKey)

	// Execute the request & read all the bytes of the response
	res, _ := http.DefaultClient.Do(req)
	defer res.Body.Close()
	body, _ := io.ReadAll(res.Body)

	if res.StatusCode != 200 {
		panic("Non-200 response: " + string(body))
	}

	// Write the bytes to a file
	file, _ := os.Create("./out/v1alpha_txt2img.png")
	defer file.Close()
	_, err := file.Write(body)
	if err != nil {
		panic(err)
	}
}
